src/blue_cwl/me_model/staging.py

```diff
@@ -382,17 +385,19 @@ def stage_emodel_configuration(
     mechanisms_dir = create_dir(Path(staging_dir, "mechanisms"))
     morphologies_dir = create_dir(Path(staging_dir, "morphology"))
 
-    # stage swc morphology path
-    staged_dataset["morphology"]["path"] = download_distribution(
-        dataset["morphology"]["id"],
-        output_dir=morphologies_dir,
-        encoding_format="application/swc",
-        base=base,
-        org=org,
-        proj=proj,
-        token=token,
-    )
-    del staged_dataset["morphology"]["id"]
+    if morphology_id := staged_dataset["morphology"].pop("id", None):
+        # stage swc morphology path
+        staged_dataset["morphology"]["path"] = download_distribution(
+            morphology_id,
+            output_dir=morphologies_dir,
+            encoding_format="application/swc",
+            base=base,
+            org=org,
+            proj=proj,
+            token=token,
+        )
+    else:
+        raise CWLWorkflowError(f"No morphology id in: {dataset['morphology']}")
```

Comment:
Given that you have access to the `entity_id` here, it would be helpful if the EModelConfiguration id is also stated in the error message.